### PR TITLE
fix: proper error handling for lnurl withdraw

### PR DESCRIPTION
**Pull request by pavanjoshi914**
### Describe the changes you have made in this PR

proper error handling for lnurl withdraw

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Screenshots of the changes [optional]

image

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/app/screens/LNURLWithdraw/index.tsx [hunk 1]
```diff
@@ -83,12 +83,16 @@ function LNURLWithdraw() {
...
+        throw new Error(response.data.reason);
       }
     } catch (e) {
       console.error(e);
-      if (e instanceof Error) {
-        toast.error(e.message);
+      if (axios.isAxiosError(e)) {
+        const error =
+          (e.response?.data as { reason?: string })?.reason || e.message;
+        toast.error(`Error: ${error}`);
```
Comment (reneaaron) [inline src/app/screens/LNURLWithdraw/index.tsx:93]:
```suggestion
        toast.error(error);
```

File: src/app/screens/LNURLWithdraw/index.tsx [hunk 2]
```diff
@@ -83,12 +83,16 @@ function LNURLWithdraw() {
...
+      } else if (e instanceof Error) {
+        toast.error(`Error: ${e.message}`);
```
Comment (reneaaron) [inline src/app/screens/LNURLWithdraw/index.tsx:95]:
```suggestion
        toast.error(error);
```

File: src/app/screens/LNURLWithdraw/index.tsx (hunk 1 above)
Comment (reneaaron) [reply src/app/screens/LNURLWithdraw/index.tsx:93]: It's obvious it's an error (there is a red X in the toast)

**Review by reneaaron — APPROVED** [review]
utACK

File: src/app/screens/LNURLWithdraw/index.tsx (hunk 2 above)
Comment (pavanjoshi914) [reply src/app/screens/LNURLWithdraw/index.tsx]: yup! but we are representing error messages in similar fashion at the rest. just to be consistent i followed the same. thanks for the commit

**Comment by pavanjoshi914** [comment]
fixed  tests accordingly merging it

**Pull request merged**